src/shogun/kernel/GaussianARDKernel.cpp

```diff
@@ -148,3 +122,18 @@ SGMatrix<float64_t> CGaussianARDKernel::get_parameter_gradient(
 		return SGMatrix<float64_t>();
 	}
 }
+
+float64_t CGaussianARDKernel::distance(int32_t idx_a, int32_t idx_b)
+{
+	REQUIRE(lhs && rhs, "Features not set!\n")
+	
+	SGVector<float64_t> avec=((CDotFeatures *)lhs)->get_computed_dot_feature_vector(idx_a);
+	SGVector<float64_t> bvec=((CDotFeatures *)rhs)->get_computed_dot_feature_vector(idx_b);
+	
+	REQUIRE(avec.vlen==bvec.vlen, "Number of right and left hand "
+			"features must be the same\n");
+	float64_t result=0.0;
+	for (index_t i=0; i<avec.vlen; i++)
```

Comment:
@karlnapf yep. this would be useful. I think Redux module is the right place for this. Also, if using Eigen3's SIMD powered operations this would be lot faster.